src/libs/Launcher.js

```diff
@@ -368,6 +370,52 @@ export default class Launcher {
     return result
   }
 
+  /**
+   * Creates a trigger to launch a service from home application which will force the end of the current job with 'context deadline exceeded'
+   * message after 20 minutes
+   *
+   * @returns {Promise<void>}
+   */
+  async createTimeoutTrigger() {
+    const { client, job } = this.getStartContext() || {}
+
+    if (!job) {
+      throw new Error('createTimeoutTrigger: no job found in context')
+    }
+
+    const { data: timeoutTrigger } = await client.save({
+      _type: 'io.cozy.triggers',
+      worker: 'service',
+      type: '@at', // did not use @in because only @at triggers remove themselves after use
+      arguments: new Date(
+        Date.now() + constants.serviceTimeoutDuration
```

Comment:
Is is possible that the device clock is not in sync with the server? (I'm wondering if the SSL certificates would throw before this step or not)

Reply:
I used the same code as the one in banks : https://github.com/cozy/cozy-banks/blob/e39fb584e39917a5329d79762a6f16c73d79e105/src/targets/services/konnectorAlerts/createTriggerAt.js#L75

But I think if the device clock is out of sync, it won't work as expected.

No idea about the ssl certificates

Reply:
Yes it would, but @in triggers do not remove themselves after run and that is why I used @at